L1Trigger/L1TTrackMatch/plugins/DisplacedVertexProducer.cc

```diff
@@ -0,0 +1,481 @@
+        }
+
+        if (tp_i->eventId().event() == 0 && tp_j->eventId().event() == 0 && fabs(tp_i->vx() - tp_j->vx()) < 0.0001 &&
+            fabs(tp_i->vy() - tp_j->vy()) < 0.0001 && fabs(tp_i->vz() - tp_j->vz()) < 0.0001 && isHard_i && isHard_j &&
+            ((tp_i->charge() + tp_j->charge()) == 0)) {
+          isReal = true;
+        }
+      }
+
+      int inTraj =
+          calcVertex(selectedTracksWithTruth[i].first, selectedTracksWithTruth[j].first, x_dv_trk, y_dv_trk, z_dv_trk);
+      Vertex_Parameters vertex = Vertex_Parameters(
+          x_dv_trk, y_dv_trk, z_dv_trk, selectedTracksWithTruth[i].first, selectedTracksWithTruth[j].first);
+
+      if(vertex.R_T>RTMax_) continue;
+      if(vertex.R_T<RTMin_) continue;
+      
+      l1t::DisplacedTrackVertex outputVertex = l1t::DisplacedTrackVertex(selectedTracksWithTruth[i].first.index,
+                                                                         selectedTracksWithTruth[j].first.index,
+                                                                         inTraj,
+                                                                         vertex.d_T,
+                                                                         vertex.R_T,
+                                                                         vertex.cos_T,
+                                                                         vertex.delta_z,
+                                                                         vertex.x_dv,
+                                                                         vertex.y_dv,
+                                                                         vertex.z_dv,
+                                                                         vertex.openingAngle,
+                                                                         vertex.p_mag,
+                                                                         isReal);
+      
+      std::vector<float> Transformed_features = {float(selectedTracksWithTruth[i].first.pt*0.25),
+	                                         float(selectedTracksWithTruth[j].first.pt*0.25),
+                                                 selectedTracksWithTruth[i].first.eta,
+                                                 selectedTracksWithTruth[j].first.eta,
+                                                 selectedTracksWithTruth[i].first.phi,
+                                                 selectedTracksWithTruth[j].first.phi,
+                                                 selectedTracksWithTruth[i].first.d0,
+                                                 selectedTracksWithTruth[j].first.d0,
+                                                 selectedTracksWithTruth[i].first.z0,
+                                                 selectedTracksWithTruth[j].first.z0,
+                                                 float(selectedTracksWithTruth[i].first.chi2rz+0.07),
```

Comment:
what is the 0.07 addition?

Reply:
This was my hack to get around the [conifer conversion bug](https://github.com/thesps/conifer/pull/76). Shifting the quality parameters by a small amount allowed the comparisons with the cutoff values in the BDT to function properly

Reply:
The conifer XGBoost comparison issue you cited has now been merged and released in conifer 1.6. The [PR to cmsdist](https://github.com/cms-sw/cmsdist/pull/9695) is open, so eventually that change will be picked up when you `#include "conifer.h"` in CMSSW